src/sage/rings/ideal.py

```diff
@@ -376,11 +376,11 @@ def __contains__(self, x):
         EXAMPLES::
 
             sage: P.<a,b,c> = QQ[]
-            sage: I = P*[a, b]
-            sage: a + b in I
+            sage: I = P * [a, b]
+            sage: a + b in I                                                            # optional - sage.libs.singular
             True
```

Comment:
Couldn't we have feature `sage.rings.polynomial` instead of `sage.libs.singular`? I feel uncomfortable with those features `sage.libs.xxx` since the feature name reflects implementation details.

Reply:
On the other hand, the `sage.libs.singular` gives attribution of this feature to a library...

Reply:
The above code uses just polynomials (over rationals). So it is obvious that it needs `sage.rings.polynomial` if such feature exists. But to think of `sage.libs.singular`, the developer should be well aware of `singular` is used to implement polynomials in sage. This is an implementation detail, and is not obvious for some developer who has worked only in, say, combinatorics.

Reply:
Hmm. Here polynomials are already available, but to compute the inclusion in `I` needs singular. That is is the point.

Reply:
I am confused. Don't we need singular just to create `P` or `I`?

Reply:
Sage has an implementation for multivariate polynomials using dictionaries. It falls back to it when Singular is not available. The ring and also ideals can be created without Singular. Only when Gröbner bases are needed, for example for ideal membership, Singular is required.

Reply:
OK. So the existence of the tag `# optional - sage.libs.singular` in the code is incomprehensible to those that are not familiar with how sage works internally with polynomials. I think this will become a maintenance burden later.

Reply:
I think this is a documentation issue that can be solved by expanding the docstring of the feature to explain what Sage uses the library for

Reply:
Essentially we should say that singular is needed if your computation involves Groebner basis. But which computation involves Groebner basis is still not obvious to the uninitiated. 

I am okay for now. The problem, if it turns to be real, may be solved later.

Reply:
Interestingly, Sage has its own "toy" implementation of Groebner bases; but even that depends on Singular